### DISCUSSION TITLE
Code Organization [ and integrations docs ]?

### DISCUSSION THREAD

**Discussion by brucearctor** [pixeltable/pixeltable#295] [Q&A] [answered]
There is a 'working with OPENAI in the integrations section [ https://docs.pixeltable.com/docs/working-with-openai
 ], but not for openai's whisper specifically.  

Does that warrant a separate integration page?  

Otherwise, https://github.com/pixeltable/pixeltable/blob/main/pixeltable/functions/whisper.py ... should that 'just' be in the https://github.com/pixeltable/pixeltable/blob/main/pixeltable/functions/openai.py

Else, what's the community strategy for the code organization, ex: what warrants individual files vs separate files?

As I look to extend/contribute to pixeltable, I do not see a mailing list with history, nor much by way of contributing guidelines, other than --> https://github.com/pixeltable/pixeltable?tab=readme-ov-file#-contributions--feedback 

That's fine, but as an open source project, figured worth having the 'discussion' somewhere public/documented to help others.

**Comment by pierrebrunelle** [comment]
The API Reference aims to be comprehensive: https://pixeltable.github.io/pixeltable/api/functions/whisper/. 

Meanwhile, the User Documentation, for now, has a goal to showcase the core functionality to help users understand what Pixeltable is and does.

Regarding contributions and community discussion, it's a great question. Pixeltable is still a recent and new project and we aim at making it extensible so people can contribute to it. Thanks for submitting https://github.com/pixeltable/pixeltable/pull/305, we will look into it soon and aim at building a first draft of a contribution guidelines.

**Reply by brucearctor** [reply]
Iwound up making sense of it.  

`whisper.py` depends on package `openai-whisper` and `openai.py`depends on openai package.   

Although both are from openai, they are different.   Also, as another example, https://github.com/pixeltable/pixeltable/blob/main/pixeltable/functions/anthropic.py has another distinct dependency.

So, I can follow the logic of different files for different `pip install XXX` [ where possible ].

**Comment by aaron-siegel** [comment, marked as answer]
`openai.py` contains functions that map to the OpenAI API (= REST endpoints on openai.com).

`openai-whisper` is a separate open source package that runs locally and is not part of the OpenAI API; hence the different Pixeltable module. Think of `openai.py` as including "everything that's part of the OpenAI API", not necessarily "everything under the umbrella of OpenAI, the company." The operation in `whisper.py` is actually quite different since it involves running an open source model that consumes local compute resources, versus calling out to a paid external service.
